### DISCUSSION TITLE
RPC call before worker initialization

### DISCUSSION THREAD

**Discussion by fr05t1k** [roadrunner-server/roadrunner#992] [Q&A] [answered]
Hey everyone,

I'm writing my RR plugin and I'm facing an issue. I'd like to store some data to RR before the workers start receiving requests.  But, the PHP worker gets an exception during the connection establishing. I assume the RPC plugin starts after the worker initialization. 

Is there any way I can do that? or any workaround for it?

I can just run `os.Exec` fro the plugin, but I just want to make sure there no other way.

Thanks!

**Comment by rustatian** [comment, marked as answer]
Hey @fr05t1k. Just use goroutine to init workers in the `Serve` function like here: https://github.com/roadrunner-server/http/blob/master/plugin.go#L112

**Reply by fr05t1k** [reply]
thanks @rustatian  again :)

**Reply by rustatian** [reply]
np, you are very welcome 👍🏻
